File: frontend/src/types/quiz.ts
/**
 * Matches `backend/app/modules/generation/models/mc_question.py` (`MultipleChoiceQuestion`).
 * Counts: `backend/app/modules/generation/config/mc_question.py` and `frontend/src/constants/mc_question.ts`.
 */

export type QuizSource = "topic" | "file";

export interface MultipleChoiceQuestion {
  question_type: "multiple_choice";
  question: string;
  options: string[];
  correct_indices: number[];
  explanation: string;
}

export interface QuizResponse {
  questions: MultipleChoiceQuestion[];
  model_used: string;
  source: QuizSource;
  truncated?: boolean;
  cost_usd: number;
}

/** POST /api/generate/question success body (wrapper). */
export interface QuestionGenerateResponse {
  question: MultipleChoiceQuestion;
  cost_usd: number;
}

File: backend/app/modules/generation/models/quiz.py
"""Collection of MCQs produced by generation pipelines.

Not the HTTP response type; see :class:`~app.modules.generation.models.responses.QuizResponse`.
"""

from pydantic import BaseModel

from app.modules.generation.models.mc_question import MultipleChoiceQuestion


class Quiz(BaseModel):
    questions: list[MultipleChoiceQuestion]
